netcore/geode-dotnet-core.sln

```diff
@@ -0,0 +1,31 @@
+﻿
```

Comment:
CMake can generate VS project files if that is your preferred generator so why are we checking them in?

Reply:
cmake can't, as far as we can tell, generate the "new, improved" .sln and .csproj files to target .net core.  Quite a bit of online chatter about it, but nothing conclusive, and our investigation of it has not yielded any working solution.  Since the new files are _insanely_ simple, we decided not to spend further cycles on it.

Reply:
That seems odd to me since most of the work to support C# and .NET in CMake is for .NET Core but perhaps with an emphasis on not being stuck with VS build as the build toolkit. Since I should be able to build .NET Core on my Mac I shouldn't need VS Studio and build tools to do so right? Something we should certainly improve on because checking in any project files has never gone well in all of the history of software development. ;)

Reply:
Yeah, it's weird, esp given that the new format is so darned simplified.  

What we know so far about the new solution/project format:
* Mac and Linux platforms can be a little "grumpy" about the ones created in VS2019, though they ostensibly read em.  We haven't characterized the issue(s) yet, but for right now I would avoid messing with project files in VS2019.
* VS2019, JetBrains Rider, and VS Code can read, build, and test with them, on all platforms
* The preferred way to edit them appears to be `dotnet` command from the command line.  This is actually awesome.

Reply:
Sorry, forgot to answer your question: Yes, you should be able to build and run tests on Mac without any IDE, just an install of the .net 5.0 SDK.

I'm working on a BUILDING.md for .net core, but the short version is:
* Install .net 5.0 SDK and set up PATH to tools
* From $GeodeNative_ROOT/netcore:
   `dotnet build`
   `startserver.sh`
   `dotnet test`
   `stopserver.sh`

Reply:
FWIW, based on earlier comments I moved us back to .net core 3.1 until such time as we can go to v6.  So don't bother installing 5